File: src/main/java/com/cxxcxx/zinecraft/core/sound/ModSound.java
package com.cxxcxx.zinecraft.core.sound;

import com.cxxcxx.zinecraft.api.registry.builder.MusicDiscBuilder;
import com.cxxcxx.zinecraft.core.Zinecraft;
import net.minecraft.data.worldgen.BootstrapContext;
import net.minecraft.world.item.JukeboxSong;

public final class ModSound {
  public static final MusicDiscBuilder AMBIENT_PICTURES_OF_THE_PAST = musicDisc("pictures_of_the_past", 95.0F, "James Primate - Pictures of the Past");
  public static final MusicDiscBuilder AMBIENT_RANDOM_GODS = musicDisc("random_gods", 199.0F, "James Primate - Random Gods (Theme III)");
  public static final MusicDiscBuilder AMBIENT_STRANGER_THINK = musicDisc("stranger_think", 240.0F, "C418 - Stranger Think");

  private ModSound() {
  }

  private static MusicDiscBuilder musicDisc(String path, float length, String description) {
    return new MusicDiscBuilder(Zinecraft.SOUNDS, Zinecraft.ITEMS, path, length, description).build();
  }

  public static void configure(BootstrapContext<JukeboxSong> context) {
    Zinecraft.SOUNDS.bootstrapSongs(context);
  }

  public static void bootstrap() {
  }
}

File: src/main/kotlin/com/cxxcxx/zinecraft/core/data/ModSound.kt
package com.cxxcxx.zinecraft.core.data

import com.cxxcxx.zinecraft.api.content.SongEntry
import com.cxxcxx.zinecraft.core.ZinecraftCore
import net.minecraft.data.worldgen.BootstrapContext
import net.minecraft.world.item.JukeboxSong

object ModSound {
  val AMBIENT_PICTURES_OF_THE_PAST = song(
    "ambient.pictures_of_the_past", 95f,
    "James Primate - Pictures of the Past"
  )
  val AMBIENT_RANDOM_GODS = song(
    "ambient.random_gods", 199f,
    "James Primate - Random Gods (Theme III)"
  )
  val AMBIENT_STRANGER_THINK = song(
    "ambient.stranger_think", 240f,
    "C418 - Stranger Think"
  )

  fun configure(context: BootstrapContext<JukeboxSong>) {
    ZinecraftCore.CONTENT.bootstrapSongs(context)
  }

  private fun song(path: String, length: Float, description: String): SongEntry =
    ZinecraftCore.CONTENT.musicDisc(path, length, description)
}
